### ISSUE TITLE
It seems that there is some weird problem in the task switch

### ISSUE THREAD

**Issue by Dreaded-Gnu** [bolthur/kernel#45]
While working on dynamic linker and system calls in #41, it appears that it's not possible any longer there to execute the vfs daemon without the optimization for size, the linker strip and gcc garbage sections.

When debugging under qemu via gdb, steps are correctly done. But as soon as the task switch switches to another task, on switch back to vfs it starts again at the entry point _start instead of continuing where it was previously.

The behaviour occurs only when building the whole project with debug symbols, which makes it even more strange as it is.

**Comment by Dreaded-Gnu** [comment]
After testing a while around it seems that it's not a bug, it's just pure slowness. So may be there could be something speed up once #41 is finished and reintegrated.

**Comment by Dreaded-Gnu** [comment]
Slowness issue has improved with changes in #41. Seems that Removal of prefer size over speed option within newlib build and some other enhancements have solved the issue.

Closed issue as duplicate of #41
